internal/app/admin/http/handler.go

```diff
@@ -102,45 +102,73 @@ func configDumpHandler(bootstrapConfig *bootstrapv1.Bootstrap) http.HandlerFunc
 	}
 }
 
-// TODO(lisalu): Support dump of matching resources when cache key regex is provided.
+func printCacheEntry(key string, cache cache.ReadOnlyCache, w http.ResponseWriter) {
+	resource, err := cache.FetchReadOnly(key)
+	if err != nil {
+		fmt.Fprintf(w, "no resource for key %s found in cache.\n", key)
+		return
+	}
+	resourceString, err := resourceToString(resource)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "unable to convert resource to string.\n")
+		return
+	}
+	fmt.Fprintf(w, "%s: %s\n", key, resourceString)
+}
+
 func cacheDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		cacheKey := getParam(req.URL.Path)
 		cache := orchestrator.Orchestrator.GetReadOnlyCache(*o)
 
 		// If no key is provided, output the entire cache.
-		if cacheKey == "" {
+		if cacheKey == "" || cacheKey == "*" {
 			keys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
 			if err != nil {
 				w.WriteHeader(http.StatusInternalServerError)
 				fmt.Fprintf(w, "error in getting cache keys: %s", err.Error())
 				return
 			}
 			for key := range keys {
-				resource, err := cache.FetchReadOnly(key)
-				if err == nil {
-					resourceString, err := resourceToString(resource)
-					if err == nil {
-						fmt.Fprintf(w, "%s: %s\n", key, resourceString)
-					}
-				}
+				printCacheEntry(key, cache, w)
 			}
 			return
 		}
 
-		// Otherwise return the cache entry corresponding to the given key.
-		resource, err := cache.FetchReadOnly(cacheKey)
-		if err != nil {
-			fmt.Fprintf(w, "no resource for key %s found in cache.\n", cacheKey)
-			return
-		}
-		resourceString, err := resourceToString(resource)
-		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			fmt.Fprintf(w, "unable to convert resource to string.\n")
+		// If wildcard suffix provided, output all cache entries that match given prefix.
```

Comment:
What do you think of the idea of combining these two if-blocks? We can detect the case of `cacheKey` being empty string and replace with the value `"*"`, then we could handle both cases with a single if-block.

Reply:
Sure, I still want to add an extra if to avoid checking for prefixes in the print total cache case to avoid unnecessary steps.